engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -278,4 +284,63 @@ public static synchronized void changeLoggersLevel(@Nonnull Map<String, String>
     }
     return ImmutableMap.copyOf(matchedClasses);
   }
+
+  /**
+   * Returns logger log file name
+   * @return logFileName
+   */
+  @Evolving
+  public static String getLogFileName()
+  {
+    Set<FileAppender> fileAppenders = getRootFileAppenders();
+    //skip fetching log file name if we have multiple file Appenders
+    if (fileAppenders.size() == 1) {
+      FileAppender fileAppender = fileAppenders.iterator().next();
+      if (isErrorLevelEnable(fileAppender)) {
+        return fileAppender.getFile();
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns offset of current log file.
+   * @return logFileOffset
+   */
+  @Evolving
+  public static long getLogFileOffset()
+  {
+    Set<FileAppender> fileAppenders = getRootFileAppenders();
+    //skip fetching log file offset if we have multiple file Appenders
+    if (fileAppenders.size() == 1) {
+      FileAppender fileAppender = fileAppenders.iterator().next();
+      if (isErrorLevelEnable(fileAppender)) {
+        File logFile = new File(fileAppender.getFile());
+        return logFile.length();
```

Comment:
My take is to disable the feature if there is no way to flush when immediateFlush is false. A few last entries including the error event may not be present in the log file.